snapshots/output/dart3-features/lib/main.dart

```diff
@@ -0,0 +1,159 @@
+  (String, int) userInfo(Map<String, dynamic> json) {
+// definition scip-dart pub dart3_features 1.0.0 lib/main.dart/
+// ^^^^^^ reference scip-dart pub dart:core 3.1.0 dart:core/string.dart/String#
+//         ^^^ reference scip-dart pub dart:core 3.1.0 dart:core/int.dart/int#
+//              ^^^^^^^^ definition scip-dart pub dart3_features 1.0.0 lib/main.dart/userInfo().
+//              documentation ```dart
+//                       ^^^ reference scip-dart pub dart:core 3.1.0 dart:core/map.dart/Map#
+//                           ^^^^^^ reference scip-dart pub dart:core 3.1.0 dart:core/string.dart/String#
+//                                            ^^^^ definition local 0
+//                                            documentation ```dart
+    return (json['name'] as String, json['height'] as int);
+//          ^^^^ reference local 0
+//                          ^^^^^^ reference scip-dart pub dart:core 3.1.0 dart:core/string.dart/String#
+//                                  ^^^^ reference local 0
+//                                                    ^^^ reference scip-dart pub dart:core 3.1.0 dart:core/int.dart/int#
+  }
+  
+  void parser(int charCode) {
+//     ^^^^^^ definition scip-dart pub dart3_features 1.0.0 lib/main.dart/parser().
+//     documentation ```dart
+//            ^^^ reference scip-dart pub dart:core 3.1.0 dart:core/int.dart/int#
+//                ^^^^^^^^ definition local 1
+//                documentation ```dart
+    const slash = 1;
+//        ^^^^^ definition local 2
+//        documentation ```dart
+    const star = 2;
+//        ^^^^ definition local 3
+//        documentation ```dart
+    const plus = 3;
+//        ^^^^ definition local 4
+//        documentation ```dart
+    const minus = 4;
+//        ^^^^^ definition local 5
+//        documentation ```dart
+  
+    final nextCharCode = 'a';
+//        ^^^^^^^^^^^^ definition local 6
+//        documentation ```dart
+    const digit0 = 0;
+//        ^^^^^^ definition local 7
+//        documentation ```dart
+    const digit9 = 9;
+//        ^^^^^^ definition local 8
+//        documentation ```dart
+  
+    void skipComment() {}
+//       ^^^^^^^^^^^ definition scip-dart pub dart3_features 1.0.0 lib/main.dart/skipComment().
+//       documentation ```dart
+    void operator(int charCode) {}
```

Comment:
TODO: rename because github's syntax highlighting isn't smart enough to know this is a funciton and not a keyword. Will improve readability when looking at code